### PR TITLE
Add Group 1, 2 & 7 data queries / indoor & outdoor unit diagnostics

### PR DESCRIPTION
**Pull request by its-tom**
Hi @mill1000,

this PR extends the AC device with support for three additional group data queries (Group 1, Group 2 and Group 7), exposing detailed diagnostics, target vs actual frequencies, indoor fan speeds, condensate pump status, and real-time power consumption, all previously unavailable through the standard `GetState` command.

> **Credit:** The group protocol research and initial implementation was done by [@TurboLed](https://github.com/TurboLed) in their fork [`TurboLed/midea-msmart@group1and2Data`](https://github.com/TurboLed/midea-msmart/tree/group1and2Data). Big thanks for reversing the payload format and sharing the work! 🙌

---

## Motivation

The standard `GetStateCommand` only returns basic operational data (temperatures, mode, fan speed level). For deeper diagnostics — compressor health, refrigerant circuit temperatures, actual measured fan speeds, power draw, and condensate pump status — the device responds to separate group polling commands.

---

### `command.py`
- **New `GetGroupCommand(group: int)`** — generic group query command replacing the hardcoded `GetGroup5Command`. Encodes `payload[3] = 0x40 | group`, supporting groups 1, 2, 4, 5 and 7.
- **Removed `GetGroup5Command`** — no longer needed; callers use `GetGroupCommand(5)`.
- **Extended response router** — `GROUP_DATA` frames are now routed to `Group1Response`, `Group2Response` or `Group7Response` based on the group byte.
- **New response classes:**

  | Class | Group | Properties |
  |---|---|---|
  | `Group1Response` | 1 | `compressor_frequency`, `target_compressor_frequency`, `compressor_current`, `compressor_voltage`, `temp_indoor_coil` (T1), `temp_evaporator` (T2), `temp_condenser` (T3), `temp_outdoor` (T4), `temp_discharge_pipe` (TP) |
  | `Group2Response` | 2 | `indoor_fan_speed`, `target_indoor_fan_speed`, `water_pump_running` |
  | `Group7Response` | 7 | `compressor_power` |

### `device.py`
- **New enable flags:** `enable_group1_data_requests`, `enable_group2_data_requests`, `enable_group7_data_requests`
- **`refresh()`** sends `GetGroupCommand(1/2/7)` when the corresponding flag is set
- **`_update_state()`** handles all three new response types
- **New read-only properties** on `AirConditioner` for all 13 new sensor values
- **`to_dict()`** includes all new fields

> **Note on naming:** Properties use `compressor_*` instead of `outdoor_unit_*` to be device-agnostic — on some units (e.g. PortaSplit) the compressor is located inside the indoor unit.

### `cli.py`
- **New `--group {1,2,5,7}` flag** for `msmart-ng query`, repeatable:
  ```bash
  msmart-ng query 192.168.1.100 --auto --group 1 --group 2 --group 7
  ```

---

## Usage

```python
device.enable_group1_data_requests = True   # compressor diagnostics
device.enable_group2_data_requests = True   # indoor fan speed & water pump
device.enable_group7_data_requests = True   # power in Watts
await device.refresh()

print(device.compressor_frequency)          # e.g. 28 Hz
print(device.target_compressor_frequency)   # e.g. 25 Hz
print(device.compressor_voltage)            # e.g. 232 V
print(device.compressor_current)            # e.g. 1 A
print(device.temp_indoor_coil)              # e.g. 20.5 °C  (T1)
print(device.temp_evaporator)               # e.g. 4.0 °C   (T2)
print(device.temp_condenser)                # e.g. 26.0 °C  (T3)
print(device.temp_outdoor)                  # e.g. 19.0 °C  (T4)
print(device.temp_discharge_pipe)           # e.g. 36 °C    (TP)
print(device.indoor_fan_speed)              # e.g. 424 RPM
print(device.target_indoor_fan_speed)       # e.g. 416 RPM
print(device.water_pump_running)            # e.g. False
print(device.compressor_power)              # e.g. 268 W
```

---

## Tests

All existing tests pass. Added **new tests**:

| File | New tests |
|---|---|
| `test_command.py` | `TestGetGroupCommand` (payload encoding, frame type), Group1/2/7 response parsing, zero-value edge cases, attribute presence check |
| `test_device.py` | `refresh()` sends correct `GetGroupCommand(n)` when flags enabled, Groups 1/2/7 disabled by default, `_update_state()` stores Group1/2/7 data correctly |

```
96 passed in 0.19s  ✅  (msmart/device/AC/)
```

---

## Backward Compatibility

- All existing APIs are unchanged
- Group 1, 2 and 7 queries are **opt-in** — no behavior change for existing integrations

---

## Example Output

```python
$ msmart-ng query 192.168.1.100 --auto --group 1 --group 2 --group 7
...
INFO:msmart.cli:Enabling Group 1 (outdoor unit performance) data requests.
INFO:msmart.cli:Enabling Group 2 (indoor fan data) data requests.
INFO:msmart.cli:Enabling Group 7 (outdoor unit power) data requests.
INFO:msmart.cli:Querying device state.
INFO:msmart.cli:Device state: {
  'power': True,
  'mode': <OperationalMode.COOL: 2>,
  'fan_speed': 1,
  'target_temperature': 18.0,
  'indoor_temperature': 21.0,
  'outdoor_temperature': 19.0,
  ...
  # Group 1 — compressor diagnostics
  'target_compressor_frequency': 25,
  'compressor_frequency': 28,
  'compressor_current': 1,
  'compressor_voltage': 228,
  'temp_indoor_coil': 21.0,
  'temp_evaporator': 4.0,
  'temp_condenser': 25.5,
  'temp_outdoor': 19.0,
  'temp_discharge_pipe': 36,
  # Group 2 — indoor fan & diagnostics
  'target_indoor_fan_speed': 416,
  'indoor_fan_speed': 424,
  'water_pump_running': False,
  # Group 7 — compressor power
  'compressor_power': 266
}
```

### CODE REVIEW

File: msmart/device/AC/device.py [hunk 1]
```diff
@@ -1230,7 +1373,22 @@ def to_dict(self) -> dict:
...
+            "outdoor_fan_speed": self.outdoor_fan_speed,
+            "target_compressor_frequency": self.target_compressor_frequency,
+            "compressor_frequency": self.compressor_frequency,
+            "compressor_current": self.compressor_current,
+            "compressor_voltage": self.compressor_voltage,
+            "temp_indoor_coil": self.temp_indoor_coil,
+            "temp_evaporator": self.temp_evaporator,
+            "temp_condenser": self.temp_condenser,
+            "temp_outdoor": self.temp_outdoor,
+            "temp_discharge_pipe": self.temp_discharge_pipe,
```
Comment (mill1000) [inline msmart/device/AC/device.py]: Can you rename these to not have "temp_" as a prefix and instead end with "_temperature".

e.g. 
indoor_coil_temperature
evaporator_temperature
Etc

**Comment by mill1000** [comment]
Thanks. Looks pretty good.

Do you know if any of these group data requests can be gated by certain device capabilities?

**Comment by mill1000** [comment]
Also, that Docker CI failure can be ignored. I don't have the right permissions setup apparently.

**Comment by its-tom** [comment]
> Thanks. Looks pretty good.
> 
> Do you know if any of these group data requests can be gated by certain device capabilities?

Thanks! I'm still refining it (that's why I set it to draft again) and seeing if I can extract any other interesting values from the protocol.

As for mapping these to device capabilities, I don't think there's anything that would make that straightforward. This data isn't displayed anywhere in the original app, which makes me think it isn't directly tied to any exposed device capabilities and probably can't be mapped that way very easily.

**Comment by mill1000** [comment]
That's fair. It's likely some advanced/engineering protocol that doesn't have much use for typical users.


Can you give me a heads up when you think it's ready?

**Comment by its-tom** [comment]
@mill1000, I found two additional fields: `target_indoor_fan_speed` and `water_pump_running`.

`water_pump_running` is probably the more interesting one, especially for PortaSplit users. Based on the protocol, it's bit 4 of byte 8 in Group 2. It appears to indicate the condensate water pump state, although it's also possible that it reflects the physical float switch (tank full) activating the pump.

From my side, everything looks good and this is ready to merge.

File: msmart/device/AC/device.py (hunk 1 above)
Comment (its-tom) [reply msmart/device/AC/device.py]: Done!
